### PR TITLE
SRCH-2552 Add case_sensitive option to uniqueness validations

### PR DESCRIPTION
**Pull request by lorawoodford**
## Summary
Adds `:case_sensitive` option to uniqueness validations, thus removing hundreds of the following types of deprecation warnings:

> DEPRECATION WARNING: Uniqueness validator will no longer enforce case sensitive comparison in Rails 6.1. To continue case sensitive comparison on the :channel_id attribute in YoutubeProfile model, pass `case_sensitive: true` option explicitly to the uniqueness validator.

Also includes very minor rubocop-related code cleanup in touched models (e.g. adding `frozen_string_literal`s where missing, breaking up long lines, line spacing, updating to hash syntax, etc.).
 
### Checklist
Please ensure you have addressed all concerns below before marking a PR "ready for review" or before requesting a re-review:

#### Functionality Checks

- [x] Code is functional.

- [x] Automated checks pass, if applicable. If Code Climate checks do not pass, explain reason for failures:

- [x] If your changes will be tested manually, you have run `bundle update` and committed your changes to Gemfile.lock. N/A
 
- [x] You have merged the latest changes from the target branch (usually `master` or `main`) into your branch.
 
- [x] If your target branch is NOT `master` or `main`, specify the reason: Ongoing precursor work necessary for Rails 6.1 upgrade is to use the feature/rails_6_1 branch.
 
- [x] PR title is either of the format **SRCH-#### \<description\>** matching the associated Jira ticket (i.e. "SRCH-123 implement feature X"), or **Release #.#.#** matching the release number
 
- [x] You have squashed your commits into a single commit (exception: your PR includes commits with formatting-only changes, such as required by Rubocop)

- [x] Your primary commit message is of the format **SRCH-#### \<description\>** matching the associated Jira ticket
 
#### Process Checks

- [x] You have specified an "Assignee", and if necessary, additional reviewers

### CODE REVIEW

File: spec/models/sitemap_spec.rb [hunk 1]
```diff
@@ -15,14 +17,14 @@
...
+      before { described_class.create!(valid_attributes) }
 
       it 'rejects duplicate urls' do
-        expect(described_class.new(valid_attributes)).to_not be_valid
+        expect(described_class.new(valid_attributes)).not_to be_valid
       end
 
       it 'is not case-sensitive' do
-        expect(described_class.create!(url: url.upcase)).to be_valid
+        expect(described_class.new({ url: url.upcase })).not_to be_valid
```
Comment (MothOnMars) [inline spec/models/sitemap_spec.rb:27]: For tidiness, we can omit the curly brackets:
```ruby
expect(described_class.new(url: url.upcase)).not_to be_valid
```

**Review by MothOnMars — CHANGES_REQUESTED** [review]
one tweak

File: spec/models/sitemap_spec.rb (hunk 1 above)
Comment (lorawoodford) [reply spec/models/sitemap_spec.rb]: Change made.

**Comment by MothOnMars** [comment]
LGTM

**Pull request merged**